modules/utxo-lib/test/bitgo/psbt/psbtUtil.ts

```diff
@@ -0,0 +1,278 @@
+    }
+  } else {
+    assert.deepStrictEqual(txParsed.scriptType, psbtParsed?.scriptType);
+    assert.deepStrictEqual(txParsed.pubScript, psbtParsed?.pubScript);
+
+    if (
+      (txParsed.scriptType === 'p2sh' && psbtParsed?.scriptType === 'p2sh') ||
+      (txParsed.scriptType === 'p2wsh' && psbtParsed?.scriptType === 'p2wsh') ||
+      (txParsed.scriptType === 'p2shP2wsh' && psbtParsed?.scriptType === 'p2shP2wsh')
+    ) {
+      assert.deepStrictEqual(txParsed.redeemScript, psbtParsed?.redeemScript);
+      assert.deepStrictEqual(txParsed.witnessScript, psbtParsed?.witnessScript);
+    } else if (txParsed.scriptType === 'p2tr' && psbtParsed?.scriptType === 'p2tr') {
+      // To ensure script path p2tr
+      assert.deepStrictEqual(txParsed.publicKeys, psbtParsed.publicKeys);
+      const txParsedP2trScriptPath = txParsed as ParsedSignatureScriptTaprootScriptPath;
+      assert.deepStrictEqual(txParsedP2trScriptPath.controlBlock, psbtParsed?.controlBlock);
+      assert.deepStrictEqual(txParsedP2trScriptPath.scriptPathLevel, psbtParsed?.scriptPathLevel);
+      assert.deepStrictEqual(txParsedP2trScriptPath.leafVersion, psbtParsed?.leafVersion);
+    }
+  }
+}
+
+function validatePublicKeys(
+  psbtParsed: ParsedPsbt2Of3 | ParsedPsbtP2TR | undefined,
+  txParsed: ParsedSignatureScript2Of3 | ParsedSignatureScriptTaproot | undefined
+) {
+  if (txParsed === undefined) {
+    assert.deepStrictEqual(psbtParsed?.publicKeys.length, 3);
+    psbtParsed?.publicKeys.forEach((publicKey) => {
+      assert.deepStrictEqual(Buffer.isBuffer(publicKey), true);
+    });
+  } else {
+    assert.deepStrictEqual(txParsed.publicKeys.length, psbtParsed?.publicKeys?.length);
+    const pubKeyMatch = txParsed.publicKeys.every((txPubKey) =>
+      psbtParsed?.publicKeys?.some((psbtPubKey) => psbtPubKey.equals(txPubKey))
+    );
+    assert.deepStrictEqual(pubKeyMatch, true);
+  }
+}
+
+function validateSignature(
+  psbtParsed: ParsedPsbt2Of3 | ParsedPsbtP2TR | undefined,
+  txParsed: ParsedSignatureScript2Of3 | ParsedSignatureScriptTaproot | undefined
+) {
+  if (txParsed === undefined) {
+    assert.deepStrictEqual(psbtParsed?.signatures, undefined);
+  } else {
+    const txSignatures = txParsed.signatures.filter(
+      (txSig) => Buffer.isBuffer(txSig) && !isPlaceholderSignature(txSig)
+    );
+    assert.deepStrictEqual(txSignatures.length, psbtParsed?.signatures?.length);
+    if (txSignatures.length < 1) {
+      return;
+    }
+    const sigMatch = txSignatures.every((txSig) =>
+      Buffer.isBuffer(txSig) ? psbtParsed?.signatures?.some((psbtSig) => psbtSig.equals(txSig)) : true
+    );
+    assert.deepStrictEqual(sigMatch, true);
+  }
+}
+
+export function validatePsbtParsing(
+  tx: UtxoTransaction<bigint>,
+  psbt: UtxoPsbt<UtxoTransaction<bigint>>,
+  unspents: WalletUnspent<bigint>[],
+  signatureTarget: SignatureTargetType
+): void {
+  unspents.forEach((u, i) => {
+    if (isWalletUnspent(u)) {
```

Comment:
```suggestion
    if (!isWalletUnspent(u)) {
      return;
    }
```
and have less indentation - but the signature demands `WalletUnspent` anyway so we should just remove this check

Reply:
Done